HealthcareApp/src/MigrationAdminPanel/Abstractions/MigrationsService.cs

```diff
@@ -0,0 +1,62 @@
+using Application.Abstractions;
+using Application.Abstractions.Decorators;
+using Domain.Constants;
+using Domain.Entities;
+
+namespace MigrationAdminPanel.Abstractions
+{
+    public abstract class MigrationsService
+    {
+        protected readonly IUserManagerDecorator<ApplicationUser> UserManager;
+        private readonly IScheduleRepository _repository;
+
+        protected MigrationsService(IUserManagerDecorator<ApplicationUser> userManager, IScheduleRepository repository)
+        {
+            UserManager = userManager;
+            _repository = repository;
+        }
+
+        public async Task MigrateData(string path)
+        {
+            try
+            {
+                var data = await ReadDataFromFile(path);
+
+                if (data == null)
+                {
+                    Console.WriteLine("Deserialization failed, migrationsData is null.");
+                    return;
+                }
+
+                var (patients, doctors) = ExtractUserData(data);
+
+                await UserManager.MigrateRangeAsync(patients, doctors);
+
+                foreach (var patient in patients)
+                {
+                    patient.UserName = patient.Email;
+                    await UserManager.AddToRoleAsync(patient, UserRolesConstants.Patient);
+                }
+
+                foreach (var doctor in doctors)
+                {
+                    doctor.UserName = doctor.Email;
+                    await UserManager.AddToRoleAsync(doctor, UserRolesConstants.Doctor);
+                    
+                    var workingHours = DefaultWorkingHours.CreateWorkingHours(doctor.Id).ToList();
+                    await _repository.CreateDefaultWorkingScheduleAsync(workingHours);
+                }
```

Comment:
Provide data consistency here. What if database gets downtime when you migrated 50% of the data - with this code your DB will be inconsistent